app/models/derived_variables/sales_log_variables.rb

```diff
@@ -15,5 +15,74 @@ def set_derived_fields!
     if mscharge_known.present? && mscharge_known.zero?
       self.mscharge = 0
     end
+    self.totchild = total_child
+    self.totadult = total_adult + total_elder
+    self.hhmemb = totchild + totadult
+    self.hhtype = household_type
+  end
+
+private
+
+  def total_elder
+    ages = [age1, age2, age3, age4, age5, age6]
+    ages.count { |x| !x.nil? && x >= 60 }
+  end
+
+  def total_child
+    relationships = [relat2, relat3, relat4, relat5, relat6]
+    relationships.count("C")
+  end
+
+  def total_adult
+    total = !age1.nil? && age1 >= 16 && age1 < 60 ? 1 : 0
+    total + (2..6).count do |i|
+      age = public_send("age#{i}")
+      relat = public_send("relat#{i}")
+      !age.nil? && ((age >= 16 && age < 18 && %w[P X].include?(relat)) || age >= 18 && age < 60)
```

Comment:
Could bracket `age >= 18 && age < 60` to make the logic clearer here? That's just personal preference though - I like to be super explicit.